cmd/minikube/cmd/start.go

```diff
@@ -156,6 +169,36 @@ func runStart(cmd *cobra.Command, args []string) {
 	clusterUp(&config)
 }
 
+// Set docker and oc cluster proxy in place
+func setProxy() {
+	if viper.IsSet("http-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTP_PROXY=%s", viper.GetString("http-proxy")))
+		clusterUpFlagSet.String(httpProxy, "", "HTTP proxy to use for master and builds")
+		viper.Set(httpProxy, viper.GetString("http-proxy"))
+		shellProxyEnv += fmt.Sprintf("http_proxy=%s ", viper.GetString("http-proxy"))
+	}
+	if viper.IsSet("https-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("HTTPS_PROXY=%s", viper.GetString("https-proxy")))
+		clusterUpFlagSet.String(httpsProxy, "", "HTTPS proxy to use for master and builds")
+		viper.Set(httpsProxy, viper.GetString("https-proxy"))
+		shellProxyEnv += fmt.Sprintf("https_proxy=%s ", viper.GetString("https-proxy"))
+	}
+	if viper.IsSet("no-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("NO_PROXY=%s,%s", updateNoProxyForDocker(),
+			                                   viper.GetString("no-proxy")))
+		clusterUpFlagSet.String(noProxyList, "", "List of hosts or subnets for which a proxy shouldn't be use")
+		viper.Set(noProxyList, viper.GetString("no-proxy"))
+		shellProxyEnv += fmt.Sprintf("no_proxy=%s", viper.GetString("no-proxy"))
+	} else if viper.IsSet("http-proxy") || viper.IsSet("https-proxy") {
+		dockerEnv = append(dockerEnv, fmt.Sprintf("NO_PROXY=%s", updateNoProxyForDocker()))
+	}
+}
+
+// update default no-proxy for docker
+func updateNoProxyForDocker() string {
+	return "localhost,127.0.0.1,172.30.1.1"
```

Comment:
I don't think this belongs to constants.go. It could be a const in this file for sure, but constants.go seems wrong. Personally I would also make it a const, but I am fine with a method as well.